app/src/components/TextInput.tsx

```diff
@@ -6,35 +6,48 @@ import "./Components.css";
 class TextInput extends EnhancedComponent<ITextInputProps, ITextInputState> {
 
     public static defaultProps: ITextInputProps = {
-        // callback: () => {return void},
+        ...EnhancedComponent.defaultProps,
+        onsubmit: (callback: () => void): void => {
+            callback();
+        },
         defaultText: "",
         text: "",
     };
 
     protected constructor(props: ITextInputProps) {
         super(props);
         this.state = {
-            // callback: props.callback,
+            ...this.state,
             defaultText: props.defaultText,
             text: props.text,
         };
+        this.updateText = this.updateText.bind(this);
+    }
+
+    private updateText(event: any): void {
+        this.setState({
+            text: event.target.value,
+        })
     }
 
     public render(): ReactNode {
         return (
-            <div> THIS IS A PLACEHOLDER </div>
+            <div>
+                <input type="text" name="text_input" placeholder={this.state.defaultText}
+                       value={this.state.text} onChange={this.updateText} />
```

Comment:
This is probably lacking a bit of styling, In the mockups there are text input fields that have rounded edges, and have higher width. See if you can add those to your optional props and put it in your styling